fdbserver/workloads/AtomicRestore.actor.cpp

```diff
@@ -103,7 +108,8 @@ struct AtomicRestoreWorkload : TestWorkload {
 			                              deterministicRandom()->randomInt(0, 100),
 			                              BackupAgentBase::getDefaultTagName(),
 			                              self->backupRanges,
-			                              SERVER_KNOBS->ENABLE_ENCRYPTION,
+			                              SERVER_KNOBS->ENABLE_ENCRYPTION &&
+			                                  conf.tenantMode != TenantMode::OPTIONAL_TENANT,
```

Comment:
Yeah currently optional mode causes alot of the backup tests to timeout with encryption since most of the keys don't belong to a tenant and we go to the tenant cache. So right now we will only run encrypted snapshots if the tenant mode is required or disabled

Reply:
The tests timeout just because it's expensive to go to the tenant cache, or are they timing out because something is actually wrong with the process? It seems possibly concerning either way, since even if we don't intend to do this exact scenario ourselves I think we would still be using the tenant cache.

Reply:
Timing out because it's expensive since on each miss we do a point refresh by going to the database. The scenarios that this is currently designed for is for tenant mode required/disabled which runs fine. This is a known problem and Ata and I have discussed it so if we want to support optional we will need to discuss this further and will require a more complicated solution. Right now i just want the current backup tests to test the encryption codepath so this is more of a temporary fix

Reply:
To add more context we use the TenantEntryCache so on a cache miss we don't know whether the tenant doesn't exist or simply that our cache isn't updated. Right now we assume the latter and refresh. We had an idea of using change feeds to keep the cache updated but its out of scope for now and will see if it's needed in the future

Reply:
I'm not sure I understand why we would be having so many misses though; most of these tests will be operating entirely within one tenant that doesn't change

Reply:
Not when its optional. I can see that there are a bunch of KV pairs that don't belong to any tenant

Reply:
In those cases there is still usually only 0 or 1 tenants that don't change (and possibly a handful of tenants that remain empty).

Is every failed tenant lookup falling through to a database read? I thought the idea here was to only fall through if it was known that the cache wasn't up to date.

Reply:
We still haven't full solved the problem of knowing whether the cache is up to date or not, currently we onyl disable cache lookups when tenant mode is disabled, for all other cases it falls through to a database read. Watches is a step towards making sure the cache is updated but watches themselves are not 100% reliable so we cannot be sure on a cache miss wether the cache was up to date or not

Reply:
Discussed offline there are ways to optimize this case which may be implemented in follow up prs